src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,143 @@ void medViewContainer::removeColorIndicator(QColor color)
+            root.appendChild(layeredViewInfo);
+            layeredView->write(generatedPath);
+            workingDir.cdUp();
+
+            QString generatedGlobalPath=workingDir.canonicalPath()+"/globalMapping.xml";
+
+            QFile file(generatedGlobalPath);
+            if (file.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+            {
+                QTextStream out(&file);
+                out << doc.toString();
+                QString path = dirPath + "/" + subDirName;
+                return path;
+            }
+        }
+        else
+        {
+            displayMessageError("Failed to create new directory: " + subDirName);
+        }
+    }
+    else
+    {
+        displayMessageError("Scene saving failed");
+    }
+    return "";
+}
+
+void medViewContainer::loadScene()
+{
+    //parsing the XML file describing the scene,
+    QString fileName = QFileDialog::getOpenFileName(this, tr("Open File"),"/home",tr("XML files (*.xml)"));
+    QFile file(fileName);
+    QFileInfo fileInfo(file);
+    QDir workingDir=fileInfo.dir();
+    if(file.open(QIODevice::ReadOnly))
+    {
+        QDomDocument doc("xml");
+        if(doc.setContent(&file))
+        {
+            //doc now contains the full XML tree
+            QDomNodeList viewsNodes=doc.elementsByTagName("layeredView");
+
+            for(int i=0;i<viewsNodes.size();i++)
+            {
+                QDomElement viewElement=viewsNodes.item(i).toElement();
+                if(viewElement.isNull())
+                {
+                    printInConsole("Failed to open a view, unable to find element");
+                    continue;
+                }
+                QString path=workingDir.canonicalPath()+"/"+viewElement.attribute("path");
+                if(!QFile::exists(path))
+                {
+                    printInConsole("Failed to open a view: path " + path + " does not exist");
+                    continue;
+                }
+                QFile mappingFile(path);
+                QFileInfo mappingFileInfo(mappingFile);
+                if(!mappingFile.open(QIODevice::ReadOnly ))
+                {
+                    printInConsole("Failed to open a view, unable to open mapping file");
+                    continue;
+                }
+                QDomDocument viewInfo("xml");
+                if(viewInfo.setContent(&mappingFile))
+                {
+                    QDomNodeList layersNodes=viewInfo.elementsByTagName("layer");
+
+                    for(int j=0;j<layersNodes.size();j++)
+                    {
+                        QDomElement layerElement=layersNodes.item(j).toElement();
+                        QString fileName=layerElement.attribute("filename");
+                        open(mappingFileInfo.dir().canonicalPath()+"/"+fileName);
+
+                        medAbstractView *view = this->view();
+                        if(view)
+                            view->restoreState(&layerElement);
```

Comment:
If we don't do anything here in case there is no view, then this test could be removed altogether. The day some programming bug causes this to fail, we will have a lot of trouble finding the origin.